sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+/**
+ *
+ *  CS3283/4 storage.js
+ *  This class acts as a facade for other developers to access to the database.
+ *  The implementation is a non-SQL local storage to support the WebApp.
+ *
+ */
+
+import constants from './constants';
+
+import entitiesCreator from './create/entitiesCreator';
+
+import query from './retrieve/query';
+
+import dataModifier from './data-modifier/dataModifier';
+
+import exceptions from './exceptions';
+
+let databaseName = constants.INDEXEDDB_NAME;
+
+// dummy function to initialize createTables promises
+function initCreateTable() {
+  if (typeof Promise === 'function') {
```

Comment:
I will not be able to know when the developer first use or import my library >.< because if I put a check in the constructor, it will only be called the if user use `const something = new Database()` or something. The caller can basically be able to call any of the functions directly